### ISSUE TITLE
Move attachments out of the database

### ISSUE THREAD

**Issue by balsdorf** [eventum/eventum#251]
For large installations, keeping a large number of attachments in the database can hinder database performance and create logistical problems. I plan on changing Eventum to (optionally?) store files on the filesystem. This creates alternative logistical challenges of course but I believe it is worth it.

Open questions:
* Should Eventum have the option to keep attachments in the database?
* What directory should attachments be stored in?

Thoughts?

**Comment by glensc** [comment]
i would leave it optional, because i can do the migration later, not block update of eventum because attachment migration not completed.

about directory, new setup.php option (not the php constant), that defaults to `var/attachments`?
http://symfony.com/doc/current/quick_tour/the_architecture.html

i.e the directory should definately not be public, as need to acl access to attachments.

and as for implementation, perhaps implement storage driver subsystem, then could store attachments remotely, like aws s3, or [minio](https://www.minio.io/)

**Comment by glensc** [comment]
about storage drivers, quick find lead me to [league/flysystem](http://flysystem.thephpleague.com/), but definately there are others.

we just need to implement compatible Provider for our current Database mode.

**Comment by glensc** [comment]
created branch: https://github.com/eventum/eventum/pull/254

new dependencies: none, but suggests other packages:
```
Package operations: 1 install, 0 updates, 0 removals
  - Installing league/flysystem (1.0.28): Loading from cache
league/flysystem suggests installing league/flysystem-aws-s3-v2 (Allows you to use S3 storage with AWS SDK v2)
league/flysystem suggests installing league/flysystem-aws-s3-v3 (Allows you to use S3 storage with AWS SDK v3)
league/flysystem suggests installing league/flysystem-azure (Allows you to use Windows Azure Blob storage)
league/flysystem suggests installing league/flysystem-cached-adapter (Flysystem adapter decorator for metadata caching)
league/flysystem suggests installing league/flysystem-copy (Allows you to use Copy.com storage)
league/flysystem suggests installing league/flysystem-dropbox (Allows you to use Dropbox storage)
league/flysystem suggests installing league/flysystem-eventable-filesystem (Allows you to use EventableFilesystem)
league/flysystem suggests installing league/flysystem-rackspace (Allows you to use Rackspace Cloud Files)
league/flysystem suggests installing league/flysystem-sftp (Allows you to use SFTP server storage via phpseclib)
league/flysystem suggests installing league/flysystem-webdav (Allows you to use WebDAV storage)
league/flysystem suggests installing league/flysystem-ziparchive (Allows you to use ZipArchive adapter)
Writing lock file
```

**Comment by glensc** [comment]
#254 is merged and will appear in 3.3.0
